ext/liquid_c/extconf.rb

```diff
@@ -1,5 +1,8 @@
 require 'mkmf'
-$CFLAGS << ' -std=c99 -Wall -Werror -Wextra -Wno-unused-parameter -Wno-missing-field-initializers'
+$CFLAGS << ' -std=c99 -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers'
+if RbConfig::CONFIG['host_os'] !~ /linux/ || RUBY_VERSION >= "2.7"
```

Comment:
Note that `RUBY_VERSION >= "2.7"` could have a problem if they didn't plan on going to ruby 3.0 on the next release.  Otherwise, that check would fail if RUBY_VERSION was `"2.10"`.  You could use `Gem::Version` objects (`Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("2.7")`) to compare the versions properly.